pat/consumer_tests/step_6.py

```diff
@@ -0,0 +1,469 @@
+                report.InvocationInfo.InvocationError,
+                report.InvocationInfo.InvocationErrorMessage,
+                extra={'step': step},
+            )
+            test_results.append(False)
+        for expected, actual in zip(
+            expected_order,
+            (p.InvocationInfo.InvocationState for p in operation_result.report_parts),
+            strict=True,
+        ):
+            if expected == actual:
+                logger.info(
+                    'The answer of the SetString operation with the handle %s operation is "%s" as expected',
+                    operation.Handle,
+                    expected,
+                    extra={'step': step},
+                )
+            else:
+                logger.error(
+                    'The answer of the SetString operation with the handle %s operation is "%s", but expected "%s"',
+                    operation.Handle,
+                    actual,
+                    expected,
+                    extra={'step': step},
+                )
+                test_results.append(False)
+        if not any(
+            expected == actual
+            for expected, actual in zip(
+                expected_order,
+                (p.InvocationInfo.InvocationState for p in operation_result.report_parts),
+                strict=True,
+            )
+        ):
+            test_results.append(False)
+        else:
+            test_results.append(True)
+    return any(test_results) and all(test_results)
+
+
+def test_6e(consumer: SdcConsumer) -> bool:  # noqa: C901, PLR0912
+    """The Reference Consumer invokes SetMetricStates:
+    - Payload: 2 metric states (settings; consider alert limits)
+    - The Reference Provider immediately responds with Fin
+    - Action: The Reference Provider alters values of given metrics
+    """  # noqa: D205, D400, D415
+    step = f'{__STEP__}e'
+    operation_timeout = 5.0
+    metric_state_counts = 2
+
+    set_service: setservice.SetServiceClient = typing.cast('setservice.SetServiceClient', consumer.set_service_client)
+
+    operations_ = consumer.mdib.descriptions.NODETYPE.get(pm_qnames.SetMetricStateOperationDescriptor)
+    if not operations_:
+        logger.error('The reference provider provides no set value operations', extra={'step': step})
+        return False
+
+    metric_settings = [
+        m
+        for m in consumer.mdib.descriptions.objects
+        if m.is_metric_descriptor and m.MetricCategory == pm_types.MetricCategory.SETTING
+    ]
+    if not metric_settings or len(metric_settings) < metric_state_counts:
+        logger.error(
+            'The reference provider does not provide at least %d metrics with @MetricCategory="%s", but found %d',
+            metric_state_counts,
+            pm_types.MetricCategory.SETTING,
+            len(metric_settings),
+            extra={'step': step},
+        )
+        return False
+    test_results: list[bool] = []
+    for operation in operations_:
```

Comment:
in my opinion the OperationTarget has to be taken into account - do not randomly select a Set-Metric